yield_plugin/src/yield_plugin.cpp

```diff
@@ -600,92 +597,154 @@ namespace yield_plugin
+    double x_lead = std::max(0.0, object_downtrack - vehicle_downtrack);
+    RCLCPP_DEBUG_STREAM(nh_->get_logger(),"x_lead: " << x_lead);
 
-      if(t_ph > config_.tpmin && t_ref > t_ph)
-      {
-        tp = t_ph;
-      }
-      else if(t_ph < config_.tpmin)
-      {
-        tp = config_.tpmin;
-      }
-      else
-      {
-        tp = t_ref;
-      }
+    double goal_velocity = get_object_velocity_along_trajectory(earliest_collision_obj.velocity.twist, original_tp, earliest_collision_time);
+    RCLCPP_DEBUG_STREAM(nh_->get_logger(),"object's speed along trajectory at collision: " << goal_velocity);
 
-      RCLCPP_DEBUG_STREAM(nh_->get_logger(),"Object avoidance planning time: " << tp);
+    // roadway object position
+    double gap_time = std::max(0.0, x_lead - config_.x_gap)/initial_velocity;
 
-      update_tpp_vector = generate_JMT_trajectory(original_tp, initial_pos, goal_pos, initial_velocity, goal_velocity, tp);
+    if (goal_velocity <= config_.min_obstacle_speed){
+      RCLCPP_WARN_STREAM(nh_->get_logger(),"The obstacle is not moving, goal velocity is set to 0 from: " << goal_velocity);
+      goal_velocity = 0.0;
+    }
 
-      return update_tpp_vector;
+    // determine the safety inter-vehicle gap based on speed
+    double safety_gap = std::max(goal_velocity * gap_time, config_.x_gap);
+    if (!std::isnormal(safety_gap))
+    {
+      RCLCPP_WARN_STREAM(rclcpp::get_logger("yield_plugin"),"Detected non-normal (nan, inf, etc.) safety_gap. Making it desired safety gap configured at config_.x_gap: " << config_.x_gap);
+      safety_gap = config_.x_gap;
+    }
+    if (config_.enable_adjustable_gap)
+    {
+      // check if a digital_gap is available
+      double digital_gap = check_traj_for_digital_min_gap(original_tp);
+      RCLCPP_DEBUG_STREAM(nh_->get_logger(),"digital_gap: " << digital_gap);
+      // if a digital gap is available, it is replaced as safety gap
+      safety_gap = std::max(safety_gap, digital_gap);
+    }
+    // safety gap is implemented
+    double goal_pos = std::max(0.0, x_lead - safety_gap);
+    double initial_pos = 0.0; //relative initial position (first trajectory point)
+    double original_max_speed = max_trajectory_speed(original_tp.trajectory_points, earliest_collision_time);
+    double delta_v_max = fabs(goal_velocity - original_max_speed);
+    RCLCPP_DEBUG_STREAM(nh_->get_logger(),"delta_v_max: " << delta_v_max << ", safety_gap: " << safety_gap);
+
+    // reference time, is the maximum time available to perform object avoidance (length of a trajectory)
+    double t_ref = (rclcpp::Time(original_tp.trajectory_points[original_tp.trajectory_points.size() - 1].target_time).seconds() - plan_start_time.seconds());
+    // time required for comfortable deceleration
+    double t_ph = config_.acceleration_adjustment_factor * delta_v_max / config_.yield_max_deceleration;
+
+    // planning time for object avoidance
+    double tp = 0;
+
+    if(t_ph > config_.tpmin && t_ref > t_ph)
```

Comment:
Small nitpick, but consider reordering the arguments to `config_.tpmin < t_ph && t_ph < t_ref` to resemble the math notation _a <= x <= b_. This is slightly more readable.

Reply:
reordered